actgen/agents/directed_qnet.py

```diff
@@ -43,51 +43,39 @@ def directed_update(self, states, actions, delta_update, n_updates, baseline_qne
                 if self.optim == 'adam':
                     optimizer = torch.optim.Adam(list(self.parameters()), lr=self.lr)
 
-                # update for normal DQN
-                if self.agent_type == 'dqn':
-                    # get q_target
-                    with torch.no_grad():
+                # get q_target
+                with torch.no_grad():
+                    # update for normal DQN
+                    if self.agent_type == 'dqn':
                         original_q_values = self.forward(s.float())
-                        q_target = original_q_values.clone()
```

Comment:
I believe we do need the `clone()` method. 

I remember trying without using clone, and `q_target` is just a name reference to the same tensor. Thus, modifying `q_target` also modifies the `original_q_values`

Reply:
Ah, okay, good catch

Reply:
Oh and even worse, I had made it so `q_target` was never initialized...